receiver/splunkhecreceiver/receiver_test.go

```diff
@@ -35,13 +35,18 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/testutil"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/sharedcomponent"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/splunk"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/splunkhecreceiver/internal/mocks"
 )
 
-func assertHecSuccessResponse(t *testing.T, resp *http.Response, body any) {
+func assertHecSuccessResponse(t *testing.T, resp *http.Response, body any, ackID ...uint64) {
 	status := resp.StatusCode
 	assert.Equal(t, http.StatusOK, status)
 	assert.Equal(t, httpJSONTypeHeader, resp.Header.Get(httpContentTypeHeader))
-	assert.Equal(t, map[string]any{"code": float64(0), "text": "Success"}, body)
+	if len(ackID) > 0 {
+		assert.Equal(t, map[string]any{"code": float64(0), "text": "Success", "ackId": float64(ackID[0])}, body)
```

Comment:
I think I added the variadic variable parameter to assertHecSuccessResponse to reuse code to verify the new ackId we now expect to see in responses. Though in retrospective this seems confusing. I have since separated assertion on non-ack and ack responses into two functions